### PR TITLE
feat: angular 19

### PR DESCRIPTION
**Pull request by quentinderoubaix**

### CODE REVIEW

File: angular/demo/bootstrap/src/app/samples/accordion/playground.route.ts [hunk 1]
```diff
@@ -22,13 +21,13 @@ const undefinedAccordionConfig = getUndefinedValues(getAccordionDefaultConfig())
 	</div>`,
 })
 export default class PlaygroundComponent {
-	@ViewChild(AccordionDirective) widget!: AccordionDirective;
-	@ViewChildren(AccordionItemComponent) itemWidgets!: QueryList<AccordionItemComponent>;
+	readonly widget = viewChild.required(AccordionDirective);
+	readonly itemWidgets = viewChildren(AccordionItemComponent);
```
Comment (MarkoOleksiyenko) [inline angular/demo/bootstrap/src/app/samples/accordion/playground.route.ts:25]: should not be required judging by the previous use?

File: angular/demo/bootstrap/src/app/samples/focustrack/focustrack.route.ts [hunk 2]
```diff
@@ -42,19 +41,12 @@ import {ChangeDetectionStrategy, Component, effect, signal} from '@angular/core'
...
-		effect(
-			() => {
-				this.activeElements.update((elements) =>
-					elements.concat([{tagName: this.activeElement()?.tagName?.toLowerCase(), id: this.activeElement()?.id || undefined}]),
-				);
-			},
-			{allowSignalWrites: true},
-		);
-	}
+	public readonly activeElements = linkedSignal<Element | null, {tagName?: string; id?: string}[]>({
```
Comment (MarkoOleksiyenko) [inline angular/demo/bootstrap/src/app/samples/focustrack/focustrack.route.ts:45]: as it is in the dev preview i would not use the linkedsignal (although it is cool)

File: angular/headless/test.ts [hunk 3]
```diff
@@ -1,10 +1,9 @@
-// This file is required by vitest
-
-import 'zone.js';
-import 'zone.js/testing';
+import '@angular/compiler';
```
Comment (MarkoOleksiyenko) [inline angular/headless/test.ts:1]: is this one required? usually it's for JIT

File: angular/headless/tsconfig.spec.json [hunk 4]
```diff
@@ -0,0 +1,11 @@
+{
+	"extends": "./tsconfig.json",
+	"compilerOptions": {
+		"outDir": "./out-tsc/spec",
+		"target": "es2016",
```
Comment (MarkoOleksiyenko) [inline angular/headless/tsconfig.spec.json:5]: es2016? :)

File: package.json [hunk 5]
```diff
@@ -228,7 +229,10 @@
 			"dependencies": [
 				"prepare",
 				"./eslint-plugin:build"
-			]
+			],
+			"env": {
+				"NODE_OPTIONS": "--max-old-space-size=8192"
```
Comment (MarkoOleksiyenko) [inline package.json:234]: why do we need this one?

Comment (quentinderoubaix) [reply package.json:234]: The new version of `angular-eslint` or `typescript` was increasing the amount of memory needed to lint the project, this environment variable helps to make sure we do not fail (in local or in the CI).

File: angular/demo/bootstrap/src/app/samples/focustrack/focustrack.route.ts (hunk 2 above)
Comment (quentinderoubaix) [reply angular/demo/bootstrap/src/app/samples/focustrack/focustrack.route.ts:45]: for demos we were already using preview things, as it's not part of an actual delivery

File: angular/demo/bootstrap/src/app/samples/accordion/playground.route.ts (hunk 1 above)
Comment (quentinderoubaix) [reply angular/demo/bootstrap/src/app/samples/accordion/playground.route.ts:25]: viewChildren does not have a `required`

File: angular/headless/test.ts (hunk 3 above)
Comment (quentinderoubaix) [reply angular/headless/test.ts:1]: Indeed this is required, as described here:
https://analogjs.org/fr/docs/features/testing/vitest#setup-for-running-tests-in-the-browser

File: angular/headless/tsconfig.spec.json (hunk 4 above)
Comment (quentinderoubaix) [reply angular/headless/tsconfig.spec.json]: removed

**Review by MarkoOleksiyenko — APPROVED** [review]
lgtm, thanks for the migration effort @quentinderoubaix !

**Comment by quentinderoubaix** [comment]
thanks for the review @MarkoOleksiyenko !

**Pull request merged**